tmail-backend/jmap/extensions-es-v7/src/main/java/com/linagora/tmail/james/jmap/ESEmailAddressContactSearchEngine.java

```diff
@@ -79,22 +79,48 @@ public ESEmailAddressContactSearchEngine(ReactorElasticSearchClient client, Elas
 
     @Override
     public Publisher<EmailAddressContact> update(AccountId accountId, MailAddress mailAddress, ContactFields updatedFields) {
-        throw new NotImplementedException("Not implemented yet!");
+        if (mailAddress.equals(updatedFields.address())) {
```

Comment:
As the document ID is based on the mailAddress, if the mailAddress changes, we then need to delete and recreate a record. If only names change, we can do a normal update. Thus the if